src/frontend/renderer.js

```diff
@@ -73,10 +76,7 @@ export function render(field, encoded_descr) {
             p2 = encodePoint(p2);
 
         } else {
-            let start = (i === blueIdx) ? 9
-                      : (i === firstIdx) ? 18
-                      : (i === secondIdx) ? 27
-                      : undefined;
+            let start = (order_counter++)*9;
```

Comment:
Try to re-work the loop to be like this:
```js
for(let i = 0; i < 5; i++) {
    let boxDescr = document.createElement("p");
    let p1;
    let p2;
    p1 = {x: field.boxes[i].right, y: field.boxes[i].top};
    p2 = {x: field.boxes[i].left, y: field.boxes[i].bott};
    p1 = encodePoint(p1);
    p2 = encodePoint(p2);
    boxDescr.appendChild(document.createTextNode(field.boxColors[i] + ": (" + p1 + " " + p2 + ")"));
    boxDescr.setAttribute("class", "descr-paragraph");
    descr.appendChild(boxDescr);
}
```
From my understanding it will lead that all coordinates of all boxes will be in proper order (unshuffled).

Reply:
The modifications have been made. Your modifications were correct and it works perfectly.